hadoop/conf/hdfs/core-site.xml

```diff
@@ -6,43 +6,27 @@
 <?xml-stylesheet type="text/xsl" href="configuration.xsl"?>
 
 <configuration>
-{%- if hdfs.namenode_count == 1  %}
+
 {%- if major == '1' %}
     <property>
         <name>fs.checkpoint.dir</name>
-        <value>
-{%- for d in hdfs.local_disks -%}
-{%- if loop.last -%}
-{{ d }}/hdfs/snn
-{%- else -%}
-{{ d }}/hdfs/snn,
-{%- endif -%}
-{%- endfor -%}</value>
+        <value>{{ '/hdfs/snn,'.join(hdfs.local_disks) }}/hdfs/snn</value>
         <final>true</final>
     </property>
 {%- endif %}
 
-    <property>
-{%- if major == '1' %}
-        <name>fs.default.name</name>
-{%- else %}
-        <name>fs.defaultFS</name>
-{%- endif %}
-        <value>hdfs://{{ hdfs.namenode_host }}:{{ hdfs.namenode_port }}</value>
-    </property>
-
 <!--
     <property>
         <name>hadoop.tmp.dir</name>
         <value>{{ hdfs.tmp_dir }}</value>
     </property>
 -->
-{%- else %}
+
     <property>
-      <name>fs.defaultFS</name>
-      <value>hdfs://{{hdfs.ha_cluster_id}}</value>
+        <name>{{ 'fs.default.name' if major == '1' else 'fs.defaultFS' }}</name>
+        <value>hdfs://{{ hdfs.namenode_host+':'+hdfs.namenode_port if hdfs.namenode_count == 1 else hdfs.ha_cluster_id }}</value>
```

Comment:
This breaks wherever one has specified the namenode port as an integer - please revert the port back to being its own expression